src/lib/notificationWorker.js

```diff
@@ -50,18 +53,38 @@ class NotificationWorker {
 
   * queueNotifications (transfer, transaction) {
     const affectedAccounts = _([transfer.debits, transfer.credits])
-      .flatten().pluck('account').map((account) => this.uri.make('account', account)).value()
+      .flatten().pluck('account').value()
     affectedAccounts.push('*')
+
+    const fulfillment = yield this.Fulfillment.findByTransfer(transfer.id, { transaction })
```

Comment:
We could save some time by only looking for the fulfillment if the transfer state is `executed` or `rejected`.